main.go

```diff
@@ -132,77 +133,78 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
-	milestones := []milestoneAPI{}
-
-	strURL := []string{baseURL, projectID, "/milestones?state=active&per_page=100"}
-	URL := strings.Join(strURL, "")
-
-	m, linkHeader, err := callServer(URL, token, milestones)
-	if err != nil {
-		return m, err
-	}
-
-	for _, milestone := range milestones {
-		m[milestone.Title] = milestone.DueDate
-	}
-
-	parsedHeader := link.Parse(linkHeader)
-	for _, elem := range parsedHeader {
-		if elem.Rel == "next" {
-			m, _, err := callServer(URL, token, milestones)
-			if err != nil {
-				return m, err
-			}
-			for _, milestone := range milestones {
-				m[milestone.Title] = milestone.DueDate
-			}
-		}
-	}
+func getActiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "state=active"
+	return getMilestones(baseURL, token, projectID, state)
+}
 
-	return m, nil
+// Get and return inactive milestones
+func getInactiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "state=closed"
+    return getMilestones(baseURL, token, projectID, state)
 }
 
-func callServer(URL string, token string, milestones []milestoneAPI) (map[string]string, string, error) {
-	m := map[string]string{}
+func getMilestones(baseURL string, token string, projectID string, state string) ([]milestoneAPI, error) {
+	strURL := []string{baseURL, projectID, "/milestones?", state}
+	URL := strings.Join(strURL, "")
+	m := []milestoneAPI{}
 	client := &http.Client{}
-	var linkHeader string
 	req, err := http.NewRequest("GET", URL, nil)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
-
 	req.Header.Add("PRIVATE-TOKEN", token)
 	resp, err := client.Do(req)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
-
 	respByte, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return m, linkHeader, err
+		return m, err
 	}
-
-	json.Unmarshal(respByte, &milestones)
+	json.Unmarshal(respByte, &m)
 	defer resp.Body.Close()
-
+	var linkHeader string
 	linkHeader = resp.Header.Get("Link")
+	parsedHeader := link.Parse(linkHeader)
+	for _, elem := range parsedHeader {
```

Comment:
Same review comment as last time. You are duplicating code. Once you are requesting the milestones without linkHeader and then move into the loop. Why not make both a loop. First request doesn't take linkHeader into account as it's still nil. When one linkHeader is returned the loop continues similar to how you have it now, if no linkHeader is returned aka no new page is available break the loop.